assets/js/blocks/reviews-by-category/edit.js

```diff
@@ -0,0 +1,315 @@
+	Button,
+	Notice,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	SelectControl,
+	ToggleControl,
+	Toolbar,
+	withSpokenMessages,
+} from '@wordpress/components';
+import classNames from 'classnames';
+import { SearchListItem } from '@woocommerce/components';
+import { Fragment, RawHTML } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+import { getAdminLink } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import EditorBlock from './editor-block.js';
+import ProductCategoryControl from '../../components/product-category-control';
+import ToggleButtonControl from '../../components/toggle-button-control';
+import { IconReviewsByCategory } from '../../components/icons';
+
+const enableReviewRating = !! ( typeof wc_product_block_data !== 'undefined' && wc_product_block_data.enableReviewRating );
+const showAvatars = !! ( typeof wc_product_block_data !== 'undefined' && wc_product_block_data.showAvatars );
+
+/**
+ * Component to handle edit mode of "Reviews by Category".
+ */
+const ReviewsByCategoryEditor = ( { attributes, debouncedSpeak, setAttributes } ) => {
+	const { className, editMode, categoryIds, showReviewDate, showReviewerName, showReviewContent, showProductName } = attributes;
+
+	const getBlockControls = () => (
+		<BlockControls>
+			<Toolbar
+				controls={ [
+					{
+						icon: 'edit',
+						title: __( 'Edit' ),
+						onClick: () => setAttributes( { editMode: ! editMode } ),
+						isActive: editMode,
+					},
+				] }
+			/>
+		</BlockControls>
+	);
+
+	const renderCategoryControlItem = ( args ) => {
+		const { item, search, depth = 0 } = args;
+		const classes = [
+			'woocommerce-product-categories__item',
+		];
+		if ( search.length ) {
+			classes.push( 'is-searching' );
+		}
+		if ( depth === 0 && item.parent !== 0 ) {
+			classes.push( 'is-skip-level' );
+		}
+
+		const accessibleName = ! item.breadcrumbs.length ?
+			item.name :
+			`${ item.breadcrumbs.join( ', ' ) }, ${ item.name }`;
+
+		return (
+			<SearchListItem
+				className={ classes.join( ' ' ) }
+				{ ...args }
+				showCount
+				aria-label={ sprintf(
+					_n(
+						'%s, has %d product',
+						'%s, has %d products',
+						item.count,
+						'woo-gutenberg-products-block'
+					),
+					accessibleName,
+					item.count
+				) }
+			/>
+		);
+	};
+
+	const getInspectorControls = () => {
+		const minPerPage = 1;
+		const maxPerPage = 20;
+
+		return (
+			<InspectorControls key="inspector">
+				<PanelBody
+					title={ __( 'Category', 'woo-gutenberg-products-block' ) }
+					initialOpen={ false }
+				>
+					<ProductCategoryControl
+						selected={ attributes.categoryIds }
+						onChange={ ( value = [] ) => {
+							const ids = value.map( ( { id } ) => id );
+							setAttributes( { categoryIds: ids } );
+						} }
+						renderItem={ renderCategoryControlItem }
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Content', 'woo-gutenberg-products-block' ) }>
```

Comment:
Most of these controls seem to be the same or almost the same as in _Reviews by Product_, right? I think we should move them to an external file so they can be shared between both blocks.